File: src/database/models/UserByEmail.ts
import { Model, Schema } from "better-cassandra";
import { IUserByEmail } from "../../types";

const schema = new Schema<IUserByEmail>({
    id: {
        type: "text",
        partitionKey: false
    },
    email: {
        type: "text",
        partitionKey: true
    }
});

export default new Model("users_by_email", schema);

File: src/database/models/UserByEmail.go
package models

import "github.com/scylladb/gocqlx/v3/table"

var userByEmailTableMetadata = table.Metadata{
	Name:    "users_by_email",
	Columns: []string{"email", "id"},
	PartKey: []string{"email"},
}

var UserByEmailTable = table.New(userByEmailTableMetadata)

type UserByEmail struct {
	Email string `json:"email" db:"email"`
	ID    string `json:"id" db:"id"`
}

func (u *UserByEmail) SchemaDefinition() []string {
	return []string{
		`CREATE TABLE IF NOT EXISTS users_by_email (
            email text PRIMARY KEY,
            id bigint
        );`,
	}
}
